lib/job.js

```diff
@@ -43,6 +44,38 @@ class Job extends BaseModel {
     }
 
     /**
+     * Lazy load the list of secrets
+     * @property secrets
+     * @return {Promise}    Resolves to a list of secrets
+     */
+    get secrets() {
+        delete this.secrets;
+
+        const secretNames = hoek.reach(this.permutations, '0.secrets', { default: [] });
```

Comment:
Oh I see. It's looking for `this.permutations[0].secrets`
But why? Is it always going to be index 0?